analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/References.cs

```diff
@@ -83,6 +84,11 @@ protected override ProgramState PreProcess(ProgramState state, IPropertyReferenc
             {
                 state = state.SetOperationValue(operation, value);
             }
+            if (operation.Property.Name is nameof(Array.Length) or nameof(List<int>.Count)
```

Comment:
While this is the correct thing to do, I have doubts if this file is the correct place to do it.  I'm considering doing this magic in the rule itself, because the constraint is rule-specific.
WDYT?